cmd/aws/create.go

```diff
@@ -161,7 +156,125 @@ func ValidateProvidedFlags(gitProvider string) error {
+	_, err := stsClient.GetCallerIdentity(context.TODO(), &sts.GetCallerIdentityInput{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to validate AWS region: %w", err)
+	}
+
+	// Retrieve credentials
+	creds, err := cfg.Credentials.Retrieve(context.TODO())
+	if err != nil {
+		return nil, fmt.Errorf("failed to retrieve AWS credentials: %w", err)
+	}
+
+	return &creds, nil
+}
+
+func ValidateAMIType(cfg aws.Config, amiType, nodeType string) error {
+	ssmTypes := map[string]string{
+		"AL2_x86_64":                 "/aws/service/eks/optimized-ami/1.29/amazon-linux-2/recommended/image_id",
+		"AL2_ARM_64":                 "/aws/service/eks/optimized-ami/1.29/amazon-linux-2-arm64/recommended/image_id",
+		"BOTTLEROCKET_ARM_64":        "/aws/service/bottlerocket/aws-k8s-1.29/arm64/latest/image_id",
+		"BOTTLEROCKET_x86_64":        "/aws/service/bottlerocket/aws-k8s-1.29/x86_64/latest/image_id",
+		"BOTTLEROCKET_ARM_64_NVIDIA": "/aws/service/bottlerocket/aws-k8s-1.29-nvidia/arm64/latest/image_id",
+		"BOTTLEROCKET_x86_64_NVIDIA": "/aws/service/bottlerocket/aws-k8s-1.29-nvidia/x86_64/latest/image_id",
+	}
+
+	_, ok := ssmTypes[amiType]
+	if !ok {
+		return fmt.Errorf("not a valid ami type")
+	}
+
+	ssmClient := ssm.NewFromConfig(cfg)
+	ssmParameterName := ssmTypes[amiType]
+
+	amiID, err := GetLatestAMIFromSSM(ssmClient, ssmParameterName)
+	if err != nil {
+		return fmt.Errorf("failed to get AMI ID from SSM: %w", err)
+	}
+	log.Info().Msgf("ami type is  %s", amiType)
+
+	architecture, err := GetAMIArchitecture(cfg, amiID)
+	if err != nil {
+		return fmt.Errorf("failed to get AMI architecture: %w", err)
+	}
+
+	instanceTypes, err := GetSupportedInstanceTypes(cfg, architecture)
+	if err != nil {
+		return fmt.Errorf("failed to get supported instance types: %w", err)
+	}
+
+	for _, instanceType := range instanceTypes {
+		if instanceType == nodeType {
+			return nil
+		}
+	}
+
+	return fmt.Errorf("node type %s not supported for %s", nodeType, amiType)
+}
+
+func GetLatestAMIFromSSM(ssmClient *ssm.Client, parameterName string) (string, error) {
+	input := &ssm.GetParameterInput{
+		Name: aws.String(parameterName),
+	}
+	output, err := ssmClient.GetParameter(context.TODO(), input)
+	if err != nil {
+		return "", fmt.Errorf("failed to initialise ssm client: %w", err)
+	}
+
+	return *output.Parameter.Value, nil
+}
+
+func GetAMIArchitecture(cfg aws.Config, amiID string) (string, error) {
+	ec2Client := ec2.NewFromConfig(cfg)
+	input := &ec2.DescribeImagesInput{
+		ImageIds: []string{amiID},
+	}
+	output, err := ec2Client.DescribeImages(context.TODO(), input)
+	if err != nil {
+		return "", fmt.Errorf("failed to describe images: %w", err)
+	}
+
+	if len(output.Images) == 0 {
+		return "", fmt.Errorf("no images found for AMI ID: %s", amiID)
+	}
+
+	val := output.Images[0]
+	return string(val.Architecture), nil
+}
+
+func GetSupportedInstanceTypes(cfg aws.Config, architecture string) ([]string, error) {
+	ec2Client := ec2.NewFromConfig(cfg)
```

Comment:
Same as other ec2 client comment. Since we seem to be doing several sequential interactions with the ec2Client, it probably makes more sense to have these as methods to a struct responsible for interacting with ec2.